### ISSUE TITLE
Plugin's events which name is a superstring of another event, fire events which name is a substring of the "superstring" event fired

### ISSUE THREAD

**Issue by thtg88** [ckeditor/ckeditor5-core#171]
Hi guys, I'm trying to create a custom plugin, mainly to add images to the content of CKEditor, together with providing editing functionality, and upload to a back-end service, similarly to what CKFinder does.

I've noticed a strange behaviour on regards of events: if an event fires of which name is a superstring of an existing events name, both events get fired. This may sound a bit cryptic, so an example is probably worth more than 1000 words: I have 2 events ```file:choose``` and ```file:choose:resizedImage```, if I fire ```file:choose:resizedImage``` ```file:choose``` fires as well. This only seems to happen if a ```:``` is present in the events' names, e.g. ```file:choose-resizedImage``` and ```file:choose``` won't fire together if ```file:choose-resizedImage``` is fired.

Is this an expected behaviour?

I have an example in a GitHub repo if it helps understanding the question.

Thanks,
Marco

**Comment by Mgsy** [comment]
cc @oleq

**Comment by oleq** [comment]
@thtg88 Yes, this is expected behaviour, [check out our API documentation](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_emittermixin-EmitterMixin.html#function-listenTo). `file:choose` is a "namespace group event" firing whenever `file:choose:*` is fired.

**Comment by thtg88** [comment]
Ah cool, didn't see that part of the docs, thanks guys!

**Comment by oleq** [comment]
Maybe this should land in https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/core-editor-architecture.html#event-system-and-observables?

**Comment by Reinmar** [comment]
Agree. Let's cover this in a more distinctive place.
